src/Compilers/CSharp/Portable/BoundTree/BoundIsPatternExpression.cs

```diff
@@ -13,11 +13,13 @@ public BoundDecisionDag GetDecisionDagForLowering(CSharpCompilation compilation)
             BoundDecisionDag decisionDag = this.ReachabilityDecisionDag;
             if (decisionDag.ContainsAnySynthesizedNodes())
             {
+                bool negated = this.Pattern.IsNegated(out var innerPattern);
+                Debug.Assert(negated == this.IsNegated);
                 decisionDag = DecisionDagBuilder.CreateDecisionDagForIsPattern(
                     compilation,
                     this.Syntax,
                     this.Expression,
-                    this.Pattern,
+                    innerPattern,
```

Comment:
:memo: For other reviewers, we sometimes generate two DAGs: one for binding/analysis and one for lowering. See the condition above (`ContainsAnySynthesizedNodes()` check), which only triggers for list-patterns at the moment. 
The first DAG is produced by `Binder.MakeIsPatternExpression` and it has the proper `IsNegated` logic that was missing here.